Yoast/wpseo-woocommerce
# CS: start using YoastCS 0.5

## Summary

This PR can be summarized in the following changelog entry:
* _N/A_

## Relevant technical choices:

This PR moves this repo onto [YoastCS](https://github.com/Yoast/yoastcs) 0.5.

Please see the individual commits for details and complete documentation about the changes.

* Fixed the YoastCS dependency to the `0.5.*` range.
* Ran a `composer update` not specific to the CS update to update all dependencies.
* Renamed the ruleset file to allow for overloading it and for sorting it with other config related files.
* Tidied up the ruleset and added extensive documentation.
    For the indentation I've chosen to use tabs as that's the convention for indentation in general in the WP world.
* **All** PHP files from the plugin will now be checked, no blanket exceptions.
* Added configuration for various WPCS sniffs to increase their usefulness.
    See [WPCS Customatizable sniff properties](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/wiki/Customizable-sniff-properties) for detailed information.
* Added a few very selective exclusions for specific test situations.
* Added one temporary exclusions for a WPCS bug which will be fixed in the next WPCS release.

The last commit maintains the previous behaviour of ignoring warnings for the build testing.
At this moment, there are only two warnings left:
* The first one - more than one class found in a file - is address by PR #220.
* For the second one, a `technical-debt` issue has been created #221. 

:point_right: If we want to be truly strict about CS, I could remove the last commit and add a temporary exception for the last remaining warning.


## Test instructions

This PR can be tested by following these steps:
* Run `vendor/bin/phpcs` without arguments to see that the library code is clean (other than the above mentioned warnings).
